### PR TITLE
fix: Issue 20485

### PR DESCRIPTION
**Pull request by ejwessel**
## **Description**

1. The networks indicator on the confirmations page was showing incorrectly for networks other than Ethereum and Sepolia. If the network was anything other than those two main networks then a gray question mark would show. Not only was this unsightly, but it's confusing for the user to see a known network in the network dropdown only for it to know be recognized in the confirmation screen.
2. The improvement is to repurpose the Network Display and reuse the the NetworkPicker so that there is consistency between the selected dropdown view and the confirmations network identifier.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25518?quickstart=1)

## **Related issues**

Fixes:
   * https://github.com/MetaMask/metamask-extension/issues/20485
   * https://github.com/MetaMask/metamask-extension/issues/25447
   * https://github.com/MetaMask/metamask-extension/issues/24450

## **Manual testing steps**

1. Go to 'Send'
4. Fill out 'Send' fields
5. Click 'Continue'
6. Validate that the network dropdown (of which is disabled) matches that of the network icon within the confirmation page; same icon and the same name.

### **Before**

Screenshot 2024-06-25 at 12 08 24
Screenshot 2024-06-25 at 12 09 15

### **After**

Screenshot 2024-06-26 at 11 53 42

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/app/network-display/network-display.js [hunk 1]
```diff
@@ -1,129 +1,26 @@
...
-        'network-display--disabled': disabled,
-        'network-display--clickable': typeof onClick === 'function',
-      })}
-      labelProps={{
-        variant: TypographyVariant.H7,
-        ...labelProps,
-      }}
+    <PickerNetwork
+      className="network-display"
+      label={currentNetwork?.nickname}
```
Comment (georgewrmarshall) [inline ui/components/app/network-display/network-display.js:16]: This should fix the failing e2e tests

```suggestion
      label={currentNetwork?.nickname}
      labelProps={{ 'data-testid': 'network-display'}}
```

Screenshot 2024-06-25 at 7 18 48 PM

**Review by georgewrmarshall — COMMENTED** [review]
Looking great! We may want to add before/after screenshots to ensure no visual regressions. Made one suggestion that should fix the failing e2e tests

**Comment by ejwessel** [comment]
Before: 
Screenshot 2024-06-25 at 12 08 24
Screenshot 2024-06-25 at 12 09 15

After:
Screenshot 2024-06-26 at 11 53 42

File: ui/components/app/network-display/network-display.js [hunk 2]
```diff
@@ -1,129 +1,27 @@
...
+      labelProps={{ 'data-testid': 'network-display' }}
+      src={currentNetwork?.rpcPrefs?.imageUrl}
+      iconProps={{ display: 'none' }} // do not show the dropdown icon
```
Comment (micaelae) [inline ui/components/app/network-display/network-display.js:19]: Thoughts on making the icon smaller? It seems to be too close to the container's borders

Comment (georgewrmarshall) [reply ui/components/app/network-display/network-display.js:19]: Agree we need some design attention here. The minimum size is `AvatarNetworkSize.Xs` which is what is being used

**Review by micaelae — APPROVED** [review]

File: ui/components/app/network-display/network-display.js (hunk 2 above)
Comment (georgewrmarshall) [reply ui/components/app/network-display/network-display.js:19]: Looks like this has been updated

Screenshot 2024-06-26 at 2 20 18 PM

File: ui/components/app/network-display/network-display.js [hunk 3]
```diff
@@ -1,129 +1,27 @@
...
+      avatarNetworkProps={{ size: AvatarNetworkSize.Xs }}
```
Comment (georgewrmarshall) [inline ui/components/app/network-display/network-display.js:20]: non-blocking: not sure why this is required it's this size by default 🤔

**Review by georgewrmarshall — APPROVED** [review]
LGTM! 🚀 
- Checked NetworkDisplay in send flow ✅ 
- Checked NetworkDisplay in storybook  ✅

**Review by OGPoyraz — APPROVED** [review]

**Pull request merged**